### ISSUE TITLE
String default for FileBrowserField not working

### ISSUE THREAD

**Issue by carlosgeos** [smacker/django-filebrowser-no-grappelli#51]
Clone of https://github.com/sehmaschine/django-filebrowser/issues/358, I decided to open it here since it looks more actively maintained.

A string default for FileBrowserField does not work since it looks for `.path` on a string:
https://github.com/smacker/django-filebrowser-no-grappelli/blob/62c01cb9863f07f746a04aa6becb3920f80550c4/filebrowser/fields.py#L113-L116

Of course this fails with:

```
AttributeError: 'str' object has no attribute 'path'
```
Other examples can be found in the original issue linked.

I'll submit the same PR, but maybe there is a better solution to this problem.

**Comment by carlosgeos** [comment]
Yes that is correct, there is no problem when using sqlite. Also, if only applying one migration, it works on all databases. (this error happens on subsequent migrations)

To reproduce, add a PostgreSQL database to the settings (I set up a remote one you can use):

```python
DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.sqlite3',
        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
    },
    'other': {
        'ENGINE': 'django.db.backends.postgresql',
        'NAME': 'default',
        'USER': 'myuser',
        'PASSWORD': 'mypass',
        'HOST': '172.104.224.150',
        'PORT': '5432',
    }
}
```

and change `models.py` to reflect some change in this default string for a FileBrowseField. For example, `uploads/myfile.jpg` to `uploads/myfile_change.jpg`

makemigrations and migrate with:

```
python manage.py migrate --database=other
```

When it tries to apply the newly created migration (with the change), it will fail with:

```
filebrowser/fields.py", line 116, in get_prep_value
    return value.path
AttributeError: 'str' object has no attribute 'path'
```

---

To reproduce with this repo's testing setup, I did:

1. Add the `psycopg2-binary==2.7.7` dependency to `tests/requirements.txt`
2. Add the Postgres db to the settings.py echo area in `./manual_test_2.sh`
3. Run `./run_tests_in_env.sh`.
4. Run `./manual_test_2.sh 2.1`.
5. Get in the virtual env and get in the project.
6. Run `python manage.py migrate --database=other`. Everything works fine.
7. Change or add the default string for a FileBrowseField
8. Make migrations and migrate to the Postgres DB
9. ERROR with:

```
  File "/Users/carlos/Documents/django-filebrowser-no-grappelli/projects/fb-dj2.1/filebrowser/fields.py", line 116, in get_prep_value
    return value.path
AttributeError: 'str' object has no attribute 'path'
```

I don't know why it doesn't fail with SQLite. Also, maybe the PR submitted needs a couple more checks not just at `get_prep_value` (haven't tested yet).

But the interesting fact is that if I wipe the database and apply only ONE migration containing the info for FileBrowseField, everything works. This makes sense I think since the code checks if it already has a value: if it doesn't, it returns value (and it WORKS) and if it already does, it returns value.path (and it DOESNT WORK with str). As soon as another migration appears that modifies some FileBrowseField, it fails.

**Comment by smacker** [comment]
released: https://pypi.org/project/django-filebrowser-no-grappelli/

**Comment by carlosgeos** [comment]
Quick question: is it useful to include the `six` package here (README) ?
![screenshot_20190202_003054](https://user-images.githubusercontent.com/5300711/52155221-f5eb8380-2681-11e9-9275-744a2ef7d423.png)

Because if the package six is not available, the following message appears:

```
filebrowser/fields.py", line 4, in <module>
    from six import string_types
ImportError: No module named 'six'
```

**Comment by smacker** [comment]
I'm sorry, it's my fault. There was an unused import of six in test requirements so I missed that we don't depend on it. I have just released v3.7.7 that removes the new dependency.

Please try new version.
